internal/db/job_tracker.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// AddJob adds a new job to the job tracker.
+// It returns the job ID and an error if the job already exists or if the creation fails
+func (d *Database) AddJob(ctx context.Context, jobType string) (jobId uuid.UUID, err error) {
+	const op = errors.Op("db.AddJob")
+	if err := d.ready(); err != nil {
+		return jobId, errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	db := d.DB.WithContext(ctx)
+
+	entry := dbmodel.NewJobTrackerEntry(jobType)
+	if err := db.Create(entry).Error; err != nil {
+		err := dbError(err)
+		if errors.ErrorCode(err) == errors.CodeAlreadyExists {
+			zapctx.Debug(ctx, "job already exists", zap.String("jobID", entry.JobID.String()))
+			return jobId, errors.E(op, fmt.Sprintf("job %s already exists", entry.JobID), err)
+		}
+		return jobId, errors.E(op, err)
+	}
+
+	jobId = entry.JobID
+	return jobId, nil
+}
+
+// StopJob marks a job as stopped.
+// It returns an error if the job does not exist or if the update fails.
+func (d *Database) StopJob(ctx context.Context, jobId uuid.UUID) (err error) {
+	const op = errors.Op("db.StopJob")
+	if err := d.ready(); err != nil {
+		return errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	db := d.DB.WithContext(ctx)
+	if err := db.Model(&dbmodel.JobTrackerEntry{}).Where("job_id = ?", jobId).Update("stop", true).Error; err != nil {
```

Comment:
why not use the same pattern as for methods below?
```
entry.SetStopping()
return d.updateJobStatus(ctx, entry)
```
also, what would call this method? if the job manager is trying to stop a job, i'd expect it to set a `stopping` status, not `stopped`.